server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManagerImpl.java

```diff
@@ -388,20 +389,22 @@ private static Configuration getVolumeManagerConfiguration(AccumuloConfiguration
       final Configuration hadoopConf, final String filesystemURI) {
 
     Map<String,String> volumeHdfsConfigOverrides =
-        conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX);
+        conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
+            .stream().filter(e -> e.getKey().startsWith(filesystemURI + "."))
+            .collect(Collectors.toUnmodifiableMap(Entry::getKey, Entry::getValue));
 
     // Calling new Configuration(Configuration) will synchronize on the constructor parameter
     // and can cause issues if many threads call this method concurrently.
     if (!volumeHdfsConfigOverrides.isEmpty()) {
+
       final Configuration volumeConfig = new Configuration(hadoopConf);
 
-      volumeHdfsConfigOverrides.entrySet().stream()
-          .filter(e -> e.getKey().startsWith(filesystemURI + ".")).forEach(e -> {
-            String key = e.getKey().substring(filesystemURI.length() + 1);
-            String value = e.getValue();
-            log.info("Overriding property {} for volume {}", key, value, filesystemURI);
-            volumeConfig.set(key, value);
-          });
+      volumeHdfsConfigOverrides.entrySet().forEach(e -> {
```

Comment:
You don't have to make this change but I noticed this could be simplified slightly with something like:

```
volumeHdfsConfigOverrides.forEach((k, v) -> {
    String key = k.substring(filesystemURI.length() + 1);
    log.info("Overriding property {}={} for volume {}", key, v, filesystemURI);
    volumeConfig.set(key, v);
});
```

Reply:
Other than this small nit (which you can ignore if you want as I said) the PR looks good to me. It would be nice if we could write a test for this specific `getVolumeManagerConfiguration()` method but it's private. However, it looks like the  testConfigurationOverrides() test inside of the VolumeManagerImplTest class provides coverage on the prefix usage already so I think it's ok.

Reply:
I made your suggested change, and went ahead and moved the substring call up too in cfb3047. Yes, that test should be sufficient I think.